packages/puppeteer-core/src/node/BrowserFetcher.ts

```diff
@@ -374,25 +378,21 @@ export class BrowserFetcher {
   /**
    * @remarks
    * This method is affected by the current `product`.
-   * @returns A promise with a list of all revision strings (for the current `product`)
+   * @returns A list of all revision strings (for the current `product`)
    * available locally on disk.
    */
-  async localRevisions(): Promise<string[]> {
-    if (!existsSync(this.#downloadFolder)) {
+  localRevisions(): string[] {
+    if (!existsSync(this.#downloadPath)) {
       return [];
     }
-    const fileNames = await readdir(this.#downloadFolder);
+    const fileNames = readdirSync(this.#downloadPath);
```

Comment:
let's not use the sync functions to not block the main loop while the file is read? as it's not only used in CLI but could be part of a server app.

Reply:
We can't get away with async here because we use it synchronously to resolve the executablePath. Otherwise we would have to change executablePath to promises.

Reply:
I see that the previous version was async? do we have some new behaviour/potentially breaking change then?

Reply:
It's not breaking since someone can use `await` with a synchronous functions (it's a no-op). It would be breaking if it was the reverse.